app/assets/javascripts/discourse/models/post-stream.js.es6

```diff
@@ -453,14 +453,19 @@ const PostStream = Ember.Object.extend({
         existing = postIdentityMap.get(postId);
 
     if(existing){
-      const url = "/posts/" + postId;
-      Discourse.ajax(url).then(
+      const postUrl = "/posts/" + postId,
+            topicUrl = "/t/" + existing.topic_id;
+      Discourse.ajax(postUrl).then(
         function(p){
           self.storePost(Discourse.Post.create(p));
         },
         function(){
           self.removePosts([existing]);
         });
+      Discourse.ajax(topicUrl).then(
```

Comment:
I think so.

The main 'bug' in all of this is that when a topic is deleted, it calls `PostDestroyer.destroy(topic.posts.first)`
This notifies the client that the post has been deleted, but says nothing about the topic itself. So as far as the composer (and the rest of the clientside app) knows, the topic is just fine and dandy.

I think duplicating that logic on the client side (If a topic's first post is deleted, then the topic is deleted) is a bit fragile (and the Store is so abstracted away in this app that I couldn't find a clean way of doing it anyhow), and that writing a line that's like 'if topic.deleted || topic.posts[0].deleted' in several places is gross.

Happy to be wrong, but I didn't see a way out of calling the server again in some way to update the topic.